pkg/lokifrontend/frontend/v2/frontend.go

```diff
@@ -297,7 +304,28 @@ func (f *Frontend) Do(ctx context.Context, req queryrangebase.Request) (queryran
 		// Buffer of 1 to ensure response or error can be written to the channel
 		// even if this goroutine goes away due to client context cancellation.
 		enqueue:  make(chan enqueueResult, 1),
-		response: make(chan *frontendv2pb.QueryResultRequest, 1),
+		response: make(chan ResponseTuple, 1),
+	}
+
+	if f.cfg.Encoding == EncodingProtobuf {
+		freq.queryRequest, err = f.codec.QueryRequestWrap(ctx, req)
+		if err != nil {
+			return nil, fmt.Errorf("cannot wrap request: %w", err)
+		}
+	} else {
+		httpReq, err := f.codec.EncodeRequest(ctx, req)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert request to HTTP request: %w", err)
+		}
+
+		if err := user.InjectOrgIDIntoHTTPRequest(ctx, httpReq); err != nil {
+			return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
+		}
```

Comment:
Non-blocking:
Was surprised by this; shouldn't we automatically inject the orgID into the request when encoding it?

Reply:
Awesome catch!